### PR TITLE
Eng 18513 add eecheck test simulating mp execution

### PR DESCRIPTION
**Pull request by kaikaizi**
New ee test added that simulates interleaved txn execution with snapshot process.
Found a bug from this test of an edge condition, fixed. Also contains some minor simplifications of the module.

### CODE REVIEW

File: src/ee/storage/TableTupleAllocator.hpp [hunk 1]
```diff
@@ -451,7 +451,7 @@ namespace voltdb {
              *   the tuple that gets moved to the hole by deletion, and
              *   its content.
              */
-            void update(void const* src, void const* dst);         // src tuple from temp table written to dst in persistent storage
+            void update(void const* dst);                          // src tuple from temp table written to dst in persistent storage. src doesn't matter
```
Comment (zeemanhe) [inline src/ee/storage/TableTupleAllocator.hpp]: I wonder if update() is needed since the address of a tuple is not moved for update.

Comment (kaikaizi) [reply src/ee/storage/TableTupleAllocator.hpp]: Update needs to be registered since we need to know the value before the in-place update change (`dst` in here). We don't care about the value/position after the in-place change.
In fact, we only care about the position of post-change for the delete case.
